core/src/main/java/org/apache/iceberg/deletes/NonCounter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.deletes;
+
+/**
+ * A dummy counter that does not increment its count.
+ */
+class NonCounter implements DeleteCounter {
```

Comment:
Not sure why we need this class, Is there a case when we really don't want to count? I would imagine perf wise the difference between counting and not counting is pretty small?

Reply:
@kbendick suggested a no-op version of the `DeleteCounter` instead of using null. There are legacy uses where counting is not needed, but I tend to agree that it doesn't hurt to use a single `DeleteCounter` throughout and that `NonCounter` is probably not needed.